### PR TITLE
User api

### PR DESCRIPTION
**Pull request by magnus-thor**
### https://www.pivotaltracker.com/story/show/152632963 ###
-----------------------
- setup user model
- setup endpoint for user

### CODE REVIEW

File: spec/support/response_json.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module ResponseJSON
+  def response_json
+    JSON.parse(response.body)
+  end
+end
+
+RSpec.configure do |config|
+  config.include ResponseJSON
```
Comment (tochman) [inline spec/support/response_json.rb:8]: Please move this to `rails_helper`

File: spec/requests/api/v1/registrations_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    it 'an already registered email' do
+      FactoryBot.create(:user, email: 'example@craftacademy.se',
+      password: 'password',
+      password_confirmation: 'password')
+
+      post '/api/v1/auth', params: {
+      email: 'example@craftacademy.se', password: 'password',
+      password_confirmation: 'password'
+      }, headers: headers
```
Comment (tochman) [inline spec/requests/api/v1/registrations_spec.rb:45]: Please format the params for better readability

File: spec/requests/api/v1/registrations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      expect(response_json['errors']['email']).to eq ['is not an email']
+      expect(response.status).to eq 422
+    end
+
+    it 'an already registered email' do
+      FactoryBot.create(:user, email: 'example@craftacademy.se',
+      password: 'password',
+      password_confirmation: 'password')
```
Comment (tochman) [inline spec/requests/api/v1/registrations_spec.rb:40]: Please format the params for better readability

File: spec/requests/api/v1/registrations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          email: 'example@craftacademy', password: 'password',
+          password_confirmation: 'password'
+      }, headers: headers
+
+      expect(response_json['errors']['email']).to eq ['is not an email']
+      expect(response.status).to eq 422
+    end
+
+    it 'an already registered email' do
+      FactoryBot.create(:user, email: 'example@craftacademy.se',
```
Comment (tochman) [inline spec/requests/api/v1/registrations_spec.rb:38]: Please add ` config.include FactoryBot::Syntax::Methods` to your `rails_helper` to be able to omit the Module name in specs, like:
```
create(:user)
```

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -1,5 +1,3 @@
-require ‘coveralls’
-Coveralls.wear_merged!(‘rails’)
```
Comment (tochman) [inline spec/rails_helper.rb:2]: Why did you remove this?

File: config/initializers/devise_token_auth.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  # config.headers_names = {:'access-token' => 'access-token',
+  #                        :'client' => 'client',
+  #                        :'expiry' => 'expiry',
+  #                        :'uid' => 'uid',
+  #                        :'token-type' => 'token-type' }
+
+  # By default, only Bearer Token authentication is implemented out of the box.
+  # If, however, you wish to integrate with legacy Devise authentication, you can
+  # do so by enabling this flag. NOTE: This feature is highly experimental!
+  # config.enable_standard_devise_support = false
```
Comment (olofvi) [inline config/initializers/devise_token_auth.rb:47]: Clean up some comments

**Comment by BasilMawejje** [comment]
Great job.

**Review by olofvi — COMMENTED** [review]
Good job, just need some cleaning

**Review by tochman — COMMENTED** [review]
Some comments but I would like you to add the `ActiveModelSerializer` at this point so that you can start following the JSON:API specification on the user model.

**Comment by tochman** [comment]
Please do detailed code reviews before merging. The commit history is recorded.
